### ISSUE TITLE
On order item update "on backorder" mete is not updated

### ISSUE THREAD

**Issue by nattila1** [woocommerce/woocommerce#25937]
**Describe the bug**
When a product is allowed for backorder, on a successful order a "Backordered" meta value is saved to the order item. It contains the number of items should be backordered. In case if the item is not available from stock, but three items were ordered value of this meta will be 3, etc.
In case if the number of order items is modified in the admin during the order processing, value of this meta is not recalculated. In some cases it can cause confusion. 

Image:
![image](https://pasteboard.co/IZpfU3L.png)

**Steps to reproduce the issue**

1. Select a product and Enable stock management
2. Allow backorders, set to: Allow, but notify customer
3. Create an order with that product 
4. Edit the order in admin
5. Update, increase or decrease qty. of line the item.

**Expected behavior**
Value of "Backordered" meta should be in alinement with line item qty and stock. It should be recalculated when a line item is updated. 

**Isolating the problem**

- [x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://aeco.hu
Site address (URL): https://aeco.hu
WC Version: 3.9.1
REST API Version: ✔ 1.0.7
Log Directory Writable: ✔
WP Version: 5.3.2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: –
Language: hu_HU
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.2.22
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.65.3
OpenSSL/1.0.2s

SUHOSIN Installed: –
MySQL Version: 5.6.45-cll-lve
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.9.1
WC Database Prefix: wp_
Teljes adatbázis mérete: 28.53MB
Adatbázis adatok mérete: 24.26MB
Adatbázis index mérete: 4.27MB
wp_woocommerce_sessions: Data: 0.26MB + Index: 0.01MB + Engine MyISAM
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_order_items: Data: 0.04MB + Index: 0.03MB + Engine MyISAM
wp_woocommerce_order_itemmeta: Data: 0.27MB + Index: 0.21MB + Engine MyISAM
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_adtribes_my_conversions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_commentmeta: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_comments: Data: 0.40MB + Index: 0.19MB + Engine MyISAM
wp_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_options: Data: 2.55MB + Index: 0.10MB + Engine MyISAM
wp_postmeta: Data: 14.82MB + Index: 2.46MB + Engine MyISAM
wp_posts: Data: 5.05MB + Index: 0.47MB + Engine MyISAM
wp_social_users: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_termmeta: Data: 0.02MB + Index: 0.02MB + Engine MyISAM
wp_terms: Data: 0.01MB + Index: 0.03MB + Engine MyISAM
wp_term_relationships: Data: 0.25MB + Index: 0.52MB + Engine MyISAM
wp_term_taxonomy: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_usermeta: Data: 0.07MB + Index: 0.05MB + Engine MyISAM
wp_users: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_wc_download_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_product_meta_lookup: Data: 0.05MB + Index: 0.09MB + Engine MyISAM
wp_wc_tax_rate_classes: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_webhooks: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpml_mails: Data: 0.44MB + Index: 0.00MB + Engine MyISAM
wp_yith_wcwl: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_yith_wcwl_lists: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_yoast_seo_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_yoast_seo_meta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM

### Post Type Counts ###

attachment: 2904
blocks: 8
nav_menu_item: 45
page: 20
post: 2
postman_sent_mail: 250
product: 1020
revision: 11
scheduled-action: 88
shop_order: 224
shop_order_refund: 7
wpcf7_contact_form: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (10) ###

Árukereső megbízható bolt integráció: írta Aeco - Németh Attil – 1.0
! Integration for Billingo & WooCommerce CUstomized: írta AECO – 1.8.0 – Nincsen tesztelve a Woocommerce aktív verziójával
Contact Form 7: írta Takayuki Miyoshi – 5.1.6
! Woocommerce Foxpost Parcel Customized: írta AECO by Foxpost-GZ – 1.1.1 – Nincsen tesztelve a Woocommerce aktív verziójával
Barion Payment Gateway for WooCommerce: írta Peter Szel  – 3.1.0 – Nincsen tesztelve a Woocommerce aktív verziójával
Post SMTP: írta Yehuda Hassine – 2.0.10
Advanced Order Export For WooCommerce: írta AlgolPlus – 3.1.1 – Nincsen tesztelve a Woocommerce aktív verziójával
Product Feed PRO for WooCommerce: írta AdTribes.io – 6.4.8 – Nincsen tesztelve a Woocommerce aktív verziójával
WooCommerce: írta Automattic – 3.9.1 – 4.0.0 elérhető
WP Rocket: írta WP Media – 3.4.4

### Inactive Plugins (4) ###

Asset CleanUp: Page Speed Booster: írta Gabriel Livan – 1.3.5.4
GDPR Cookie Consent: írta WebToffee – 1.7.7
Preview E-mails for WooCommerce: írta Digamber Pradhan – 1.5.4 – Nincsen tesztelve a Woocommerce aktív verziójával
YITH WooCommerce Wishlist: írta YITH – 2.2.13 – Nincsen tesztelve a Woocommerce aktív verziójával

### Dropin Plugins (1) ###

advanced-cache.php: advanced-cache.php

### Settings ###

API Enabled: –
Force SSL: –
Currency: HUF (Ft)
Currency Position: right_space
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Webshop főoldal: ❌ Az oldal ID-ja beállítva
de az oldal nem létezik

Kosár: #66 - /cart/
Pénztár: #65 - /checkout/
A fiókom: #64 - /felhasznaloi-fiok/
Általános Szerződési Feltételek: #5212 - /altalanos-szerzodesi-feltetelek/

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.10.1
Parent Theme Author URL: http://www.uxthemes.com/
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome-child/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome-child/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome-child/woocommerce/checkout/payment.php
flatsome-child/woocommerce/checkout/review-order.php verzió 3.3.0 elavult. Az új verzió: 3.8.0
flatsome/woocommerce/checkout/terms.php
flatsome-child/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-product_cat.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome-child/woocommerce/emails/customer-completed-order.php
flatsome-child/woocommerce/emails/customer-note.php
flatsome-child/woocommerce/emails/customer-on-hold-order.php
flatsome-child/woocommerce/emails/customer-processing-order.php
flatsome-child/woocommerce/emails/email-addresses.php
flatsome-child/woocommerce/emails/email-footer.php
flatsome-child/woocommerce/emails/email-header.php
flatsome-child/woocommerce/emails/email-order-details.php
flatsome-child/woocommerce/emails/email-order-items.php
flatsome-child/woocommerce/emails/email-styles.php
flatsome-child/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php verzió 3.5.0 elavult. Az új verzió: 3.9.0
flatsome/woocommerce/notices/notice.php verzió 3.5.0 elavult. Az új verzió: 3.9.0
flatsome/woocommerce/notices/success.php verzió 3.5.0 elavult. Az új verzió: 3.9.0
flatsome-child/woocommerce/order/order-details-item.php
flatsome-child/woocommerce/order/order-details.php
flatsome/woocommerce/product-searchform.php
flatsome-child/woocommerce/single-product/add-to-cart/simple.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome-child/woocommerce/single-product/product-attributes.php verzió 3.1.0 elavult. Az új verzió: 3.6.0
flatsome-child/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome-child/woocommerce/single-product/rating.php
flatsome-child/woocommerce/single-product/related.php verzió 3.0.0 elavult. Az új verzió: 3.9.0
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome-child/woocommerce/single-product/up-sells.php
flatsome-child/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php

Outdated Templates: ❌
					
					
						Hogyan kell frissíteni

### Action Scheduler ###

Teljesítve: 87
Oldest: 2020-02-14 21:45:06 +0000
Newest: 2020-03-16 19:15:07 +0000

Függőben: 0
Oldest: –
Newest: –

Megszakítva: 0
Oldest: –
Newest: –

Folyamatban: 0
Oldest: –
Newest: –

Sikertelen: 1
Oldest: 2019-10-21 19:59:40 +0000
Newest: 2019-10-21 19:59:40 +0000

`
```
</details>

**Comment by juliaamosova** [comment]
Hi @nattila1,

Thank you for taking the time to report this bug, we really appreciate your help. I _think_ can reproduce it on my end. 

What I see is the following:

1) I created a product with zero stock, enabled stock and allowed backorders with `Allow but notify customer` setting. 

2) Next, I placed an order on the site as the customer would. 

The stock of the product got decreased by 4 - as expected:

![st1](https://user-images.githubusercontent.com/19143190/77357222-2c6cdd80-6d1e-11ea-9fd8-9b3d971214d6.jpg)

On the order page, I can see that the stock was reduced in order notes - as expected:

![st2](https://user-images.githubusercontent.com/19143190/77357355-6342f380-6d1e-11ea-8ad8-f95beccc4f8b.jpg)

The customer received an email that 4 items they ordered ar eon the backorder - as expected:  

![st3](https://user-images.githubusercontent.com/19143190/77357436-8a99c080-6d1e-11ea-9f18-72d4b79cc3e6.jpg)

3) Next, I edited the order manually on the order page from item number 4 to 6. 

The stock got updated to -6 - as expected:

![st4](https://user-images.githubusercontent.com/19143190/77357731-0431ae80-6d1f-11ea-89bc-cab26d49a4e0.jpg)

On the order page, the stock got adjusted in the order notes - as expected: 

![st5](https://user-images.githubusercontent.com/19143190/77357764-114e9d80-6d1f-11ea-9a67-822adb551a43.jpg)

Email to the customer shows item quantity as 6 but `Backordered` still shows 4 - this is the issue, correct? Would you please confirm so that we are on the same page? 

![st6](https://user-images.githubusercontent.com/19143190/77357902-4eb32b00-6d1f-11ea-809c-4729f5aed02d.jpg)

We won’t be able to include this fix in the upcoming release due to the lower priority of this issue compared to others reported. We’re going to add it to our backlog so we can include it in our planning for one of our future releases.

**Comment by nattila1** [comment]
Yes, exactly. This is the issue what I found. Value of "Backordered" meta is not updated. 
Thank you!

**Comment by juliaamosova** [comment]
@nattila1 thank you for confirming!

**Comment by decox1st** [comment]
I'm having this same issue. Please is there a fix for it? Thanks

**Comment by masteradhoc** [comment]
Im having the same issue.

**Comment by knittingtoday** [comment]
Same issue. If I modify the inventory on the product page to a positive number, the backorder message  on the existing order does not go away even though inventory is now available to ship. I can, however, edit the item and remove the backorder meta data. It seems like it should go away automatically upon adjusting inventory. And, when I edit the meta data to remove the backorder message, there is no note added for that action.

**Comment by jamesckemp** [comment]
This issue has been automatically closed as part of the backlog cleanup announced in [Refining how we manage the WooCommerce backlog](https://developer.woocommerce.com/2026/05/29/backlog-cleanup/). It met the criteria outlined there. If this still affects a current version of WooCommerce, please follow the steps in that post. A fresh report with current reproduction steps is the fastest path to triage.
